### PR TITLE
control-service: Enhance Exception Handling for DataJobsSynchronizer

### PR DESCRIPTION
**Pull request by mivanov1988**
Why
DataJobsSynchronizer does not catch ApiException.

What
We handled ApiException in DataJobsSynchronizer.

Testing done:
Unit tests.

Signed-off-by: Miroslav Ivanov miroslavi@vmware.com

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java [hunk 1]
```diff
@@ -43,15 +45,24 @@ public class DataJobsSynchronizer {
...
-        deploymentService.findAllActualDeploymentNamesFromKubernetes();
+    Set<String> dataJobDeploymentNamesFromKubernetes;
+    try {
+      dataJobDeploymentNamesFromKubernetes =
+          deploymentService.findAllActualDeploymentNamesFromKubernetes();
+    } catch (ApiException e) {
+      log.error(
+          "Skipping data job deployment synchronization because deployment names cannot be loaded"
+              + " from Kubernetes.",
+          e);
```
Comment (antoniivanov) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java:61]: Wrap this in KubernetesException. 

The ApiException message is really unhelpful - it doesn't print at all the content of the body or headers and it's very hard to understand what is the problem without those

**Review by antoniivanov — APPROVED** [review]
Aside from the comment looks good to me. Thanks.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java (hunk 1 above)
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java]: Done

**Review by dakodakov — APPROVED** [review]
LGTM

**Pull request merged**